### PR TITLE
SDC tols

### PR DESCRIPTION
**Pull request by zingale**
## PR summary

This splits the tolerance that the nonlinear solve for SDC reactions uses to have separate values for density, species, and energy.

## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite
- [ ] all functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
blocked by #585

**Comment by zingale** [comment]
this also now implements an absolute tolerance that helps a lot with species integration

**Comment by zingale** [comment]
reacting_convergence still gets 4th order with these changes

**Comment by zingale** [comment]
this is ready for review.  I am happy with the convergence with these changes.

File: Source/driver/sdc_util.F90 [hunk 1]
```diff
@@ -248,8 +265,22 @@ subroutine sdc_solve(dt_m, U_old, U_new, C, sdc_iteration)
...
        endif
 
+       ! relative tolerances
+       rtol(0) = tol_dens
+       rtol(1:nspec_evolve) = tol_spec
+       rtol(nspec_evolve+1) = tol_ener
+
+       ! absolute tolerances
+       atol(0) = sdc_solver_atol * U_old(URHO)
+       atol(1:nspec_evolve) = sdc_solver_atol * U_old(URHO)   ! this way, atol is the minimum x
```
Comment (maxpkatz) [inline Source/driver/sdc_util.F90]: I don't follow this choice or the comment -- can you elaborate?

Comment (zingale) [reply Source/driver/sdc_util.F90]: we are integrating (rho X), so I need to turn what I want the minimum value of X to track into the atol for (rho X).

Comment (maxpkatz) [reply Source/driver/sdc_util.F90]: OK. But since sum(rho X) = rho, isn't this scaling based on X == 1?

Comment (maxpkatz) [reply Source/driver/sdc_util.F90]: OK, I get it now.

**Pull request merged**